main/inc/lib/usermanager.lib.php

```diff
@@ -5005,4 +5005,47 @@ public static function getUsersByOfficialCode($officialCode)
         }
         return $users;
     }
+
+    /**
+     * Calc the expended time (hh::mm:ss) by a user in a course
+     * @param int $userId The user id
+     * @param string $courseCode The course id
+     * @param int $sessionId Optional. The session id
+     * @param string $from Optional. From date
+     * @param string $until Optional. Until date
+     * @return string The time
+     */
+    public static function getExpendedTimeInCourses($userId, $courseCode, $sessionId = 0, $from = '', $until = '')
+    {
+        $userId = intval($userId);
+        $sessionId = intval($sessionId);
+
+        $trackCourseAccessTable = Database::get_statistic_table(TABLE_STATISTIC_TRACK_E_COURSE_ACCESS);
+
+        $whereConditions = array(
+            'user_id = ? ' => $userId,
+            "AND course_code = '?' " => $courseCode,
+            'AND session_id = ? ' => $sessionId
+        );
+
+        if (!empty($from) && !empty($until)) {
+            $whereConditions["AND (login_course_date >= '?' "] = $from;
+            $whereConditions["AND logout_course_date <= DATE_ADD('?', INTERVAL 1 DAY)) "] = $until;
+        }
+
+        $trackResult = Database::select(
+            'SEC_TO_TIME(SUM(UNIX_TIMESTAMP(logout_course_date) - UNIX_TIMESTAMP(login_course_date))) as total_time',
```

Comment:
I think that SEC_TO_TIME is mysql specific. It should be better to do it in PHP.

There's a function that does something like this in  main/newscorm  "calculateScormTime()"
Maybe it should be a good idea to create a function that does this in main_api.lib